Makefile

```diff
@@ -74,26 +74,43 @@ endif
 # export LSAN_OPTIONS=suppressions=.asanignore
 AFLAGS = -fsanitize=address #-fsanitize=undefined -fno-omit-frame-pointer
 
-CC_VENDOR := $(firstword $(filter gcc clang,$(shell $(CC) --version)))
+CC_VENDOR := $(firstword $(filter gcc clang icc XL,$(shell $(CC) --version)))
+FC_VENDOR := $(firstword $(filter GNU ifort XL,$(shell $(FC) --version 2>&1 || $(FC) -qversion)))
+
+# Default extra flags by vendor
+MARCHFLAG.gcc           := -march=native
+MARCHFLAG.clang         := $(MARCHFLAG.gcc)
+MARCHFLAG.icc           :=
+OMP_SIMD_FLAG.gcc       := -fopenmp-simd
+OMP_SIMD_FLAG.clang     := $(OMP_SIMD_FLAG.gcc)
+OMP_SIMD_FLAG.icc       := -qopenmp-simd
+OPT.gcc                 := -ffp-contract=fast
+OPT.clang               := $(OPT.gcc)
+CFLAGS.gcc              := -fPIC -std=c99 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CFLAGS.clang            := $(CFLAGS.gcc)
+CFLAGS.icc              := $(CFLAGS.gcc)
+CFLAGS.XL               := -qpic -MMD
+CXXFLAGS.gcc            := -fPIC -std=c++11 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CXXFLAGS.clang          := $(CXXFLAGS.gcc)
+CXXFLAGS.icc            := $(CXXFLAGS.gcc)
+CXXFLAGS.XL             := -qpic -MMD
+FFLAGS.GNU              := -fPIC -cpp -Wall -Wextra -Wno-unused-parameter -Wno-unused-dummy-argument -MMD -MP
+FFLAGS.ifort            := -fPIC -cpp
+FFLAGS.XL               := -qpic -ffree-form -qpreprocess -qextname -MMD
 
 cc_check_flag = $(shell $(CC) -E $(1) -x c /dev/null > /dev/null 2>&1 && echo 1)
```

Comment:
It looks like `xlc` does not fail if the option is not supported -- probably the preprocessor simply passes them to next stage if it does no recognize them.

Reply:
Yeah, so I think we have to rely on the vendor defaults written here or the user to pass suitable `CFLAGS`, etc. We could write a real configure that works around the apparent thicket of compiler bugs, but I'm dubious of the value and it's not for v0.6.

Reply:
If we leave it "as is" then it will be good to leave a comment here that the check does not actually work with some compilers, e.g. `xlc`.

Reply:
Another relatively easy option is to compile or even compile+link some trivial *.c file, instead of just running the preprocessor.

Reply:
I pushed the docs. Actually compiling gets slow with vendor compilers and creates potential race conditions, so I think it's the purview of an actual configure.